spec/solver_spec.rb

```diff
@@ -0,0 +1,30 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Testing Solver Class' do
+    it 'Should return the factorial of number passed to factorial method' do
+      solver = Solver.new
+      expect(solver.factorial(5)).to eq(120)
+      expect(solver.factorial(6)).to eq(720)
+      expect(solver.factorial(7)).to eq(5040)
+    end
+  end
```

Comment:
- [ ] Hi you really did a great job. As a requirement, please kindly test your factorial with a negative number and it should return an exception. If possible also test it with `0` and it should return `1` as a result